### DISCUSSION TITLE
[ v3.20.0 ] 破坏性更新 / Breaking Changes

### DISCUSSION THREAD

**Discussion by xhofe** [AlistGo/alist#4702] [Announcements]
#### 配置文件结构更新：
- 将根中的`address`, `port`, `https_port`移动到scheme中
- 移除了scheme中的`disable_http`与`https`,改为由端口号判断，为-1时则表示不启用对应协议

#### 我需要如何修改配置文件？
- 如果你修改过根中的`address`, `port`，那么你需要在scheme中再次修改一次，如果没有则不需要
- 如果你启用了https，那么你需要在scheme中修改`https_port`以重新启用
- 如果你禁用了http，那么你需要在scheme中修改`http_port`为`-1`以重新禁用

**Comment by xhofe** [comment]
#### Configuration file structure update:
- Move `address`, `port`, `https_port` in root to scheme
- Removed `disable_http` and `https` in the scheme, changed to judge by the port number, when it is -1, it means that the corresponding protocol is disabled
#### How do I modify the configuration file?
- If you have modified `address`, `port` in the root, then you need to modify it again in the scheme, if not, you don’t need it
- If you enabled https, then you need to modify `https_port` in scheme to re-enable now
- If you disabled http, then you need to modify `http_port` to `-1` in the scheme to re-disable now

**Comment by AspirantH** [comment]
大佬，我更新了这个版本，但是依然和上版一样，在luci-app-alist中的https配置都无法生效，哪怕改/etc/alist里的config也是一样，一旦在openwrt里重启了alist，配置文件会自动改回https port = -1

**Reply by xhofe** [reply]
luci-app-alist 我不了解，你需要问对应仓库的作者，我在本地测试了是正常的
![image](https://github.com/alist-org/alist/assets/36558727/98005843-cc55-43f0-af85-960ef3b97632)

**Reply by AspirantH** [reply]
谢谢回复，我再找一下原因在哪

**Reply by chengnan049** [reply]
> 谢谢回复，我再找一下原因在哪

这个双协议是被特地改了的，作者觉得没用，看他仓库的 Issues 自己编译一个

**Comment by AspirantH** [comment]
始终百思不得其解，最终还是改为默认5244端口吧

**Reply by anwen-anyi** [reply]
现在解决了吗？  使用 v3.20.1 版本

**Reply by AspirantH** [reply]
没有，尝试无果

**Reply by anwen-anyi** [reply]
> 没有，尝试无果

更新 3.20.1

**Comment by AspirantH** [comment]
今天再次升级了一下,可以了

**Reply by anwen-anyi** [reply]
因为 3.20 有一个小小的Bug 加上`luci-app-alist`配置文件也没同步更新导致的  
3.20.1修复了再加上`luci-app-alist`同步更新了配置文件自然可以啦~

**Reply by AspirantH** [reply]
谢谢大神耐心讲解~

**Reply by DanielEcho** [reply]
更新luci1.10没有
